src/appleseed/renderer/modeling/scene/assembly.cpp

```diff
@@ -56,6 +56,8 @@ namespace renderer
 // Assembly class implementation.
 //
 
+const char* Model = "assembly";
```

Comment:
> And indeed it would make sense to rename Assembly to GenericAssembly if we start to have many kinds of assemblies.

Ideally, we would have an abstract Assembly class and the current Assembly class would be a subclass of it, called GenericAssembly. But I'm not sure what would be the real advantage of doing it and in exchange we break source compatibility. I'd like to be really sure before going in 
this direction. What about renaming the current class to GenericAssembly and having a typedef GenericAssembly Assembly?